terraform/cloudwatch.tf.json.template.py

```diff
@@ -349,24 +349,27 @@ def prod_qualified_resource_name(name: str) -> str:
                                 )
                             ]
                         },
-                        'waf_rate_blocked': {
-                            'alarm_name': config.qualified_resource_name('waf_rate_blocked'),
-                            'comparison_operator': 'GreaterThanThreshold',
-                            'threshold': 0,
-                            'datapoints_to_alarm': 1,
-                            'evaluation_periods': 1,
-                            'period': 5 * 60,
-                            'metric_name': 'BlockedRequests',
-                            'namespace': 'AWS/WAFV2',
-                            'statistic': 'Sum',
-                            'treat_missing_data': 'notBreaching',
-                            'dimensions': {
-                                'WebACL': '${aws_wafv2_web_acl.api_gateway.name}',
-                                'Region': config.region,
-                                'Rule': config.waf_rate_rule_name
-                            },
-                            'alarm_actions': ['${data.aws_sns_topic.monitoring.arn}'],
-                            'ok_actions': ['${data.aws_sns_topic.monitoring.arn}'],
+                        **{
+                            f'waf_{rule_type}_blocked': {
+                                'alarm_name': config.qualified_resource_name(f'waf_{rule_type}_blocked'),
+                                'comparison_operator': 'GreaterThanThreshold',
+                                'threshold': 0,
+                                'datapoints_to_alarm': 1,
+                                'evaluation_periods': 1,
+                                'period': 5 * 60,
+                                'metric_name': 'BlockedRequests',
+                                'namespace': 'AWS/WAFV2',
+                                'statistic': 'Sum',
+                                'treat_missing_data': 'notBreaching',
+                                'dimensions': {
+                                    'WebACL': '${aws_wafv2_web_acl.api_gateway.name}',
+                                    'Region': config.region,
+                                    'Rule': getattr(config, f'waf_{rule_type}_rule_name')
```

Comment:
`config` doesn't have the attribute your are trying to access here. That raises the question why this error wasn't found when you tested this.

Reply:
This wasn't caught in testing due to adding the `fixup` commit and running IT from a project directory that had `AZUL_ENABLE_MONITORING` commented out, causing this section of `cloudwatch.tf.json.template.py` to be skipped.